Source/WTF/wtf/StructDump.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+#pragma once
+
+#include <wtf/PrintStream.h>
+
+WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+
+namespace WTF {
+
+template<typename T>
+class StructDump {
+public:
+    StructDump(const T& data)
+        : m_data(data)
+    {
+    }
+
+    void dump(PrintStream& out) const
+    {
+#if COMPILER(CLANG)
+        __builtin_dump_struct(&m_data,
+            [&]<typename ...Args>(Args&& ...args) {
+                // While the format string is not a literal, this is a string
+                // generated by clang, and clang appropriately pass a format string
+                // to this function.
+IGNORE_WARNINGS_BEGIN("format-nonliteral")
+IGNORE_WARNINGS_BEGIN("format-security")
+                out.printf(std::forward<Args>(args)...);
+IGNORE_WARNINGS_END
+IGNORE_WARNINGS_END
+            });
+#else
+        out.print("<struct>");
```

Comment:
Maybe we should do a `pointerDump()` for GCC?

Reply:
That sounds nice! Changed.